### PR TITLE
Make sure component version is always present.

### PR DESCRIPTION
**Pull request by iocanel**
- Ensure version presence in all scenarios.
- Modify integration tests to also check for version.

### CODE REVIEW

**Comment by cmoulliard** [comment]
LGTM. I will test the PR on the project `component-operator-demo` and will communicate the result

**Comment by cmoulliard** [comment]
There is a problem as the version added corresponds to the pom version of the current project but not at all to the version of the Spring Boot application. @iocanel 

Generated
```
---
apiVersion: "v1"
kind: "List"
items:
- apiVersion: "component.k8s.io/v1alpha1"
  kind: "Component"
  metadata:
    name: "fruit-client-sb"
  spec:
    runtime: spring-boot
    version: "0.0.1-SNAPSHOT"
```

Should be

```
---
apiVersion: "v1"
kind: "List"
items:
- apiVersion: "component.k8s.io/v1alpha1"
  kind: "Component"
  metadata:
    name: "fruit-client-sb"
  spec:
    runtime: spring-boot
    version: 1.5.19.RELEASE
```

**Comment by iocanel** [comment]
@cmoulliard: I see, I didn't realize how its meant to be used.

**Comment by iocanel** [comment]
@cmoulliard: it's should be fine now.

File: frameworks/spring-boot/src/main/java/io/ap4k/spring/configurator/SetSpringBootVersion.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  public static final ConfigKey<String> RUNTIME_VERSION = new ConfigKey<>("RUNTIME_VERSION", String.class);
+
+  @Override
+  public void visit(ConfigurationFluent<?> config) {
+    config.addToAttributes(RUNTIME_VERSION, getSpringBootVersion());
+  }
+
+  public static String getSpringBootVersion() {
+    try {
+      return Class.forName("org.springframework.boot.autoconfigure.SpringBootApplication")
```
Comment (geoand) [inline frameworks/spring-boot/src/main/java/io/ap4k/spring/configurator/SetSpringBootVersion.java]: Now that is seriously cool, :100:

**Pull request merged**